cairo/ethereum/cancun/vm/precompiled_contracts/alt_bn128.cairo

```diff
@@ -0,0 +1,135 @@
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin, KeccakBuiltin
+from starkware.cairo.common.math_cmp import is_le_felt
+from ethereum.cancun.vm import Evm, EvmImpl
+from ethereum.exceptions import EthereumException
+from ethereum.cancun.vm.exceptions import OutOfGasError
+from ethereum.utils.numeric import ceil32, divmod, U256_from_be_bytes
+from ethereum.cancun.vm.gas import GasConstants, charge_gas
+from ethereum_types.numeric import Uint, U256, U256Struct
+from ethereum_types.bytes import Bytes
+from ethereum.cancun.vm.memory import buffer_read
+
+func alt_bn128_add{
+    range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr: KeccakBuiltin*, evm: Evm
+}() -> EthereumException* {
+    alloc_locals;
+    let data = evm.value.message.value.data;
+
+    // Gas
+    let gas_cost = Uint(150);
+    let err = charge_gas(gas_cost);
+    if (cast(err, felt) != 0) {
+        return err;
+    }
+
+    // Operation
```

Comment:
either all follow the pattern or none
Also the natspec is present on only one function